src/test/ui/intrinsics/is_const_eval.rs

```diff
@@ -0,0 +1,11 @@
+// run-pass
+
+#![feature(core_intrinsics)]
+use core::intrinsics::is_const_eval;
+
+fn main() {
+    const X: bool = unsafe { is_const_eval() };
+    let y = unsafe { is_const_eval() };
```

Comment:
Before merging,  we need a scheme so that const prop won't prop this call. Without such a scheme in place, we'll nix the intrinsic once we support const propping function calls. cc @wesleywiser

Reply:
Where is const propagation applied to function calls ? Maybe we can just query the function path, and if it is `core::intrinsic::is_const_eval` just don't propagate it?

Reply:
It isn't yet, but that's not even the problem, the problem is propagating functions that call this internally..  or maybe that's not a problem, this is expected behaviour after all... just propagating the intrinsic itself is a problem..

Ah, we can just never prop unsafe functions.. that should do it

Reply:
@oli-obk so without updating miri in this PR, these two tests will pass ? I'm not sure how that could work since the constant evaluator won't know how to execute the first `is_const_eval` in the `const X`, right ?

Reply:
> Ah, we can just never prop unsafe functions.. that should do it

That feels too coarse unless this is the only intrinsic affected.

Reply:
Miri the tool has zero to do with const eval ;)

You want `librustc_mir/interpret/intrinsics` or whereever the size_of and friends are implemented